### PR TITLE
Update cyres_optee to use new PTA_CYRES interface

### PR DESCRIPTION
**Pull request by jordanrh1**
Update cyres_optee.c to consume new PTA_CYRES interface
https://github.com/ms-iot/optee_os/pull/59

### CODE REVIEW

**Comment by jordanrh1** [comment]
@jiria Can you point me to instructions to build and test this code?

**Comment by jiria** [comment]
https://github.com/Microsoft/openenclave/blob/feature.new_platforms/new_platforms/docs/linux_arm_dev.md

The code here will exercise it, but so will the tests:

https://github.com/Microsoft/openenclave/tree/feature.new_platforms/samples/local_attestation (note that the doc for the local_attestation is not updated and does not directly reflect the code (it is a fork from the main project)). To run it, just start one host and it will spin up on its own two TAs.

As for tests, those might be even better:

https://github.com/Microsoft/openenclave/tree/feature.new_platforms/new_platforms/tests/oetests (run the host).

Thanks,
Jirka

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

________________________________

**Comment by jordanrh1** [comment]
The build script isn't picking up `TA_DEV_KIT_DIR`. How can I build just oetests?

File: new_platforms/src/enc/optee/cyres_optee.c [hunk 1]
```diff
@@ -97,9 +66,18 @@ static TEE_Result get_pta_allocated_buf(
         oe_result = OE_OUT_OF_MEMORY;
         goto done;
     }
-    memset(local_buf, 0, local_buf_size);
```
Comment (jiria) [inline new_platforms/src/enc/optee/cyres_optee.c]: Why do we not need this?

Comment (jordanrh1) [reply new_platforms/src/enc/optee/cyres_optee.c]: PTA_CYRES returns a null-terminated string.

**Comment by jordanrh1** [comment]
Update submodule.

**Pull request closed without merging**